cpp/ql/lib/ext/std.array.model.yml

```diff
@@ -0,0 +1,12 @@
+extensions:
+  - addsTo:
+      pack: codeql/cpp-all
+      extensible: summaryModel
+    data: # namespace, type, subtypes, name, signature, ext, input, output, kind, provenance
+      - ["std", "array", True, "at", "", "", "Argument[-1].Element[*@]", "ReturnValue[*@]", "value", "manual"]
+      - ["std", "array", True, "begin", "", "", "Argument[-1].Element[*@]", "ReturnValue.Element[*@]", "value", "manual"]
+      - ["std", "array", True, "cbegin", "", "", "Argument[-1].Element[*@]", "ReturnValue.Element[*@]", "value", "manual"]
+      - ["std", "array", True, "data", "", "", "Argument[-1].Element[*@]", "ReturnValue[*@]", "value", "manual"]
+      - ["std", "array", True, "operator[]", "", "", "Argument[-1].Element[*@]", "ReturnValue[*@]", "value", "manual"]
+      - ["std", "array", True, "rbegin", "", "", "Argument[-1].Element[*@]", "ReturnValue.Element[*@]", "value", "manual"]
+      - ["std", "array", True, "rcbegin", "", "", "Argument[-1].Element[*@]", "ReturnValue.Element[*@]", "value", "manual"]
```

Comment:
This is probably the last thing you should fix (after other changes), but all of the `std` models included `bsl` equivalents as well.

Reply:
Good point. I've added these in 2ad8704325fe740f0ae43b987ae9d86993732fe9. I didn't find any `bsl` version of iterators, so I've left those out for now.

Reply:
Did another DCA run after https://github.com/github/codeql/commit/2ad8704325fe740f0ae43b987ae9d86993732fe9, and results are unchanged

Reply:
:+1:

The `StdIterator`, `IteratorTraits` and `IteratorByTypedefs` QL classes do cover `bsl`.  None of these classes were removed by your changes, however.